Fix build with GCC 4.4

Patch taken from Fedora

Original author : Lubomir Rintel <lkundrak@v3.sk>

--- eboard-1.1.1/ntext.cc.gcc44	2009-03-04 19:11:34.000000000 +0100
+++ eboard-1.1.1/ntext.cc	2009-03-04 20:04:20.000000000 +0100
@@ -244,8 +244,7 @@
     return;
   }
 
-  p = strchr(text, '\n');
-  if (p!=NULL) {
+  if (strchr(text, '\n')) {
     *p = 0;
     i = strlen(text);
     nl = new NLine(text, color);
